src/gui/osutils/macutils/AppKitImpl.mm

```diff
@@ -181,28 +184,36 @@ - (bool) enableAccessibility
 //
 // Check if screen recording is enabled, may show an popup asking for permissions
 //
-- (bool) enableScreenRecording
+- (bool) enableScreenRecording 
 {
-#if __clang_major__ >= 9 && MAC_OS_X_VERSION_MIN_REQUIRED >= 1080
-    if (@available(macOS 10.15, *)) {
-        // Request screen recording permission on macOS 10.15+
-        // This is necessary to get the current window title
-        CGDisplayStreamRef stream = CGDisplayStreamCreate(CGMainDisplayID(), 1, 1, kCVPixelFormatType_32BGRA, nil,
-                                                          ^(CGDisplayStreamFrameStatus status, uint64_t displayTime,
-                                                                  IOSurfaceRef frameSurface, CGDisplayStreamUpdateRef updateRef) {
-                                                              Q_UNUSED(status);
-                                                              Q_UNUSED(displayTime);
-                                                              Q_UNUSED(frameSurface);
-                                                              Q_UNUSED(updateRef);
-                                                          });
-        if (stream) {
-            CFRelease(stream);
-        } else {
-            return NO;
-        }
+#if __clang_major__ >= 13 && MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_VERSION_12_3
+    if (@available(macOS 12.3, *)) {
+        __block BOOL hasPermission = NO;
+        dispatch_semaphore_t sema = dispatch_semaphore_create(0);
+
+        // Attempt to use SCShareableContent to check for screen recording permission
+        [SCShareableContent getShareableContentWithCompletionHandler:^(SCShareableContent * _Nullable content, 
+                                                                        NSError * _Nullable error) {
```

Comment:
You must use `Q_UNUSED(error)` for the second parameter. Otherwise the code will not compile.

Reply:
It's compilable; otherwise, I wouldn't be able to test if the screen recording permission request dialog works properly. Additionally, the automated tests didn't report any errors. However, if you believe that adding Q_UNUSED(error) can improve the code's robustness, then it can certainly be included.

Reply:
Q_UNUSED is only useful on Linux where GCC and Clang complain of unused parameters.

Reply:
have you just disable the old permission or complete remove it? In my case macOS Sequoia just disable is not enough, I need to remove the keepassxc from privacy setting, then I see the dialog, after permit, the keepassxc is added to the list again.

<img width="461" alt="image" src="https://github.com/user-attachments/assets/d2758ebc-0934-4334-9ca5-3ab240f6ec97">

Reply:
If just disabling the setting does not work properly then there must be something missing?

Reply:
it's Macos problem, I have same story of some screenshot apps, I think it's about signature of software maybe....

Reply:
I just tested it, for the current official release 2.7.9, the behavior is the same, just disabling the setting does not work.